src/test/java/com/checkmarx/flow/cucumber/component/webhook/WebHookSteps.java

```diff
@@ -2,105 +2,177 @@
 
 import com.checkmarx.flow.CxFlowApplication;
 import com.checkmarx.flow.config.GitHubProperties;
-import cucumber.api.PendingException;
+import com.checkmarx.flow.cucumber.common.utils.TestUtils;
 import io.cucumber.java.en.Given;
 import io.cucumber.java.en.Then;
 import io.cucumber.java.en.When;
+import org.apache.commons.io.IOUtils;
 import org.apache.commons.lang3.time.StopWatch;
 import org.awaitility.Awaitility;
 import org.junit.Assert;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.SpringApplication;
-
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Optional;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestTemplate;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import javax.xml.bind.DatatypeConverter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.time.Duration;
+import java.util.*;
 import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
 
+@SpringBootTest
 public class WebHookSteps {
-    private final static Logger logger = LoggerFactory.getLogger(WebHookSteps.class);
+    private static final Charset DEFAULT_CHARSET = StandardCharsets.UTF_8;
+    private static final String WEBHOOK_REQUEST_RESOURCE_PATH = "sample-webhook-requests/from-github.json";
+    private static final int MAX_TOTAL_REQUESTS = 20;
+
+    private static final Logger logger = LoggerFactory.getLogger(WebHookSteps.class);
+    private static final Duration maxAwaitTimeForAllRequests = Duration.ofSeconds(2 * MAX_TOTAL_REQUESTS);
+
     private final List<CompletableFuture<Long>> requestSendingTasks = new ArrayList<>();
-    private final GitHubProperties properties;
 
-    public WebHookSteps(GitHubProperties properties) {
-        this.properties = properties;
-    }
+    @Autowired
+    private GitHubProperties properties;
+
+    private HttpEntity<String> webHookRequest;
+    private String cxFlowPort;
 
     @Given("CxFlow is running as a service")
     public void runAsService() {
-        SpringApplication.run(CxFlowApplication.class, "--web");
+        ConfigurableApplicationContext context = SpringApplication.run(CxFlowApplication.class, "--web");
+        cxFlowPort = context.getEnvironment().getProperty("server.port");
```

Comment:
run as a service logic should be exported to a utility

Reply:
Done: TestUtils.runCxFlowAsService()